File: core-examples/src/main/java/io/vertx/example/core/eventbus_pointtopoint/Sender.java
package io.vertx.example.core.eventbus_pointtopoint;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.eventbus.EventBus;

/*
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class Sender extends AbstractVerticle {

  @Override
  public void start() throws Exception {
    EventBus eb = vertx.eventBus();

    // Send a message every second

    vertx.setPeriodic(1000, v -> {
      eb.send("ping-address", "ping!", reply -> {
        System.out.println("Received reply " + reply.result().body());
      });
    });
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/eventbus/ssl/Sender.kt
package io.vertx.example.core.eventbus.ssl


class Sender : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    var eb = vertx.eventBus()

    // Send a message every second

    vertx.setPeriodic(1000, { v ->

      eb.request<Any>("ping-address", "ping!", { ar ->
        if (ar.succeeded()) {
          println("Received reply ${ar.result().body()}")
        } else {
          println("No reply")
        }
      })

    })
  }
}
